src/player.cpp

```diff
@@ -159,6 +160,61 @@ void Player::clearHud()
 	}
 }
 
+u32 PlayerControl::getKeysPressed() const
+{
+	u32 keypress_bits =
+		( (u32)(jump  & 1) << 4) |
+		( (u32)(aux1  & 1) << 5) |
+		( (u32)(sneak & 1) << 6) |
+		( (u32)(dig   & 1) << 7) |
+		( (u32)(place & 1) << 8) |
+		( (u32)(zoom  & 1) << 9)
+	;
+
+	// If any direction keys are pressed pass those through
+	if (direction_keys != 0)
+	{
+		keypress_bits |= direction_keys;
+	}
+	// Otherwise set direction keys based on joystick movement (for mod compatibility)
+	else if (movement_speed > 0.001f)
+	{
+		float abs_d;
+
+		// (absolute value indicates forward / backward)
+		abs_d = abs(movement_direction);
+		if (abs_d < 3.0f / 8.0f * M_PI)
+			keypress_bits |= (u32)1; // Forward
+		if (abs_d > 5.0f / 8.0f * M_PI)
+			keypress_bits |= (u32)1 << 1; // Backward
+
+		// rotate entire coordinate system by 90 degree
+		abs_d = movement_direction + M_PI_2;
+		if (abs_d >= M_PI)
+			abs_d -= 2 * M_PI;
+		abs_d = abs(abs_d);
+		// (value now indicates left / right)
+		if (abs_d < 3.0f / 8.0f * M_PI)
+			keypress_bits |= (u32)1 << 2; // Left
+		if (abs_d > 5.0f / 8.0f * M_PI)
+			keypress_bits |= (u32)1 << 3; // Right
+	}
+
+	return keypress_bits;
+}
+
+void PlayerControl::unpackKeysPressed(u32 keypress_bits)
+{
+	// It would be possible to unpack the direction keys from the bits
+	// but this is handeled differently.
```

Comment:
Too vague description. Why can't everything be updated in the same function for consistency?

Reply:
<i>It just is</i>.
I'll look at dropping `keyPressed` entirely next time I look at this PR.